tests/system/python/conftest.py

```diff
@@ -212,6 +215,138 @@ def clone_make_install():
 
     return _add_fledge_north
 
+@pytest.fixture
+def add_service():
+    def _add_service(fledge_url, service, service_branch, retries, installation_type = "make", service_name = "play",
+                     use_pip_cache = True, enabled = True):
+        
+        """ 
+            Fixture to add Service and start the start service by default 
+            service: Service to be installed
+            service_branch: Branch of service to be installed
+            retries: Number of tries for polling
+            installation_type: Type of installation for service i.e. make or package
+            service_name: Name that will be given to service to be installed
+            use_pip_cache: use pip cache in requirements installation
+            enabled: Flag to enable or disable notification instace
+        """
+        
+        # Check if the service is already installed installed
+        retval = utils.get_request(fledge_url, "/fledge/service")
+        for ele in retval["services"]:
+            if ele["type"].lower() == service:
+                return ele
+        
+        PROJECT_ROOT = Path(__file__).parent.parent.parent.parent
+        
+        # Install Service
+        def clone_make_install():
+            try:
+                subprocess.run(["{}/tests/system/python/scripts/install_c_service {} {}".format(
+                    PROJECT_ROOT, service_branch, service)], shell=True, check=True)
+            except subprocess.CalledProcessError:
+                assert False, "{} service  installation failed".format(service)
+                
+        if installation_type == 'make':
+            clone_make_install()
+        elif installation_type == 'package':
+            try:
+                subprocess.run(["sudo {} install -y fledge-service-{}".format(pytest.PKG_MGR, service)], shell=True,
+                            check=True)
+            except subprocess.CalledProcessError:
+                assert False, "{} package installation failed!".format(service)
+        else:
+            return("Skipped {} service installation. Installation mechanism is set to {}.".format(service, installation_type))
+        
+        # Add Service
+        data = {"name": "{}".format(service_name), "type": "{}".format(service), "enabled": enabled}
+        retval = utils.post_request(fledge_url, "/fledge/service", data)
+        assert service_name == retval["name"]
+        return retval
+        
+    return _add_service
+
+@pytest.fixture
+def add_notification_instance():
+    def _add_notification_instance(fledge_url, delivery_plugin, delivery_branch , rule_config={}, delivery_config={}, 
+                                   rule_plugin="Threshold", rule_branch=None, rule_plugin_discovery_name=None, 
+                                   delivery_plugin_discovery_name=None, installation_type='make', notification_type="one shot",
```

Comment:
rule_branch None only if rule_plugin exists in Threshold/DataAvailability.
And for rule and delivery plugin discovery names cannot be None in any case.